tests/integration/main.rs

```diff
@@ -67,8 +65,57 @@ fn test_expected_file(name: &str) {
         assert_eq!(lines[0..1000], expected_lines[0..1000]);
         assert!(lines == expected_lines);
     }
+}
 
-    assert_eq!(total_row_count, objects_count as u64);
+/// Converts binary columns into lists of UInt8 so they can be JSON-encoded
+fn binaries_to_uint8_list(batch: RecordBatch) -> RecordBatch {
+    RecordBatch::try_new(
+        Arc::new(Schema::new(
+            batch
+                .schema()
+                .fields
+                .into_iter()
+                .map(|field| {
+                    Field::new(
+                        field.name(),
+                        match field.data_type() {
+                            DataType::Binary => DataType::List(Arc::new(Field::new(
+                                "value",
+                                DataType::UInt8,
+                                false,
+                            ))),
+                            DataType::LargeBinary => DataType::LargeList(Arc::new(Field::new(
+                                "value",
+                                DataType::UInt8,
+                                false,
+                            ))),
+                            data_type => data_type.clone(),
+                        },
+                        field.is_nullable(),
+                    )
+                })
+                .collect::<Vec<_>>(),
+        )),
+        batch
+            .columns()
+            .iter()
+            .map(|array| match array.as_binary_opt() {
+                Some(array) => {
+                    let (offsets, values, nulls) = array.clone().into_parts();
+                    ListArray::try_new(
+                        Arc::new(Field::new("value", DataType::UInt8, false)),
+                        offsets,
+                        Arc::new(PrimitiveArray::<UInt8Type>::new(values.into(), None)),
+                        nulls,
+                    )
+                    .map(Arc::new)
+                    .expect("Could not create ListArray")
+                }
+                None => array.clone(),
+            })
+            .collect(),
+    )
+    .expect("Could not rebuild RecordBatch")
```

Comment:
I wonder if there's a way to use cast kernels to achieve the same effect instead of recreating a new recordbatch :thinking: